### DISCUSSION TITLE
How to specify order keyword in resample:  ds.resample('time='1D').interpolate(kind='polynomial',order=2)

### DISCUSSION THREAD

**Discussion by axelschweiger** [pydata/xarray#7380] [Q&A]
I am trying to use polynomial interpolation to resample an xarray in time. It says it wants the "order" keyword specfied but I don't know how. Above ds.resample('time='1D').interpolate(kind='polynomial',order=2) generates an unexpected keyword argument error. 
Thanks

**Comment by nicrie** [comment]
Are you trying to downscale or upscale your time series? I believe the operation you're trying to perform is not well defined mathematically. Perhaps I am misinterpreting what you're trying to do. But in my understanding there are two cases for the `resample` operation of a time series:
1) upscaling to a higher frequency
2) dowscaling to a lower frequency

I think there's a problem for upscaling though. Upscaling using linear interpolation works fine as you can take any two data points and fit a line between them to fill in the gaps. But how should that work for a higher polynomial, e.g. of second order? There are indefinitely many polynomials you can fit having just 2 points, so there is no unique solution. One could say, well, let's take at least three points, but then there is no guarantee that the polynomial fit of three points will smoothly fit the polynomial fit of the next following three points. In that case, what would your final time series look like?

EDIT: That being said, if you want to downscale the following may do what you're looking for?
```py
ds.interp_like(ds.resample(time='1D').mean(), method='quadratic')
```

**Reply by axelschweiger** [reply]
Thanks much Nic. 
Yes, i agree some way to specify the number of points (window size) to use in the interpolation would be useful/necessary (for both upscaling and downscaling).  I was assuming it was going to default to some minimum. At any rate if "polynomial" isn't available then something is wrong with the docs or the error messages since I can't specify the order keyword.  I think a polynomial interpolation could be useful for some processes. 

Strangely, using interpolate_na, I can specify "polynomial" as the method and even though the doc says the order keyword is required, it lets me get away without specifying it. Haven't really worked through an actual problem though to see what is happening.
